modules/lang-painless/src/test/java/org/elasticsearch/painless/WhenThingsGoWrongTests.java

```diff
@@ -197,4 +201,13 @@ public void testDynamicListWrongIndex() {
             exec("def x = new ArrayList(); x.add('foo'); return x['bogus'];");
         });
     }
+
+    public void testRCurlyNotDelim() {
+        ScriptException e = expectThrows(ScriptException.class, () -> {
+            // We don't want PICKY here so we get the normal error message
+            exec("def i = 1} return 1", emptyMap(), emptyMap(), null);
+        });
```

Comment:
note you can do expectScriptThrows, which works just like expectThrows except it unboxes ScriptException automatically. This makes it easier to assert you really got the exception you want.

e.g.:
```
IllegalArgumentException e = expectScriptThrows(IllegalArgumentException.class, ....);
```

Behind the scenes, expectScriptThrows first looks for a ScriptException (it will throw a nice error if its not that), then unboxes it and acts like expectThrows from there.

I assume in this case, the fact we have PICKY on means you get a crazy error like NoViableAlt, but its not actually complaining about ambiguity? I can look into improving PICKY potentially here, we don't want it lenient, and we want it on as much as possible.

Sadly, in antlrv4, PICKY is the only way to detect ambiguities in the grammar: You cannot do it at compile time, and antlr just defaults to being slow (bugs like #18398). It means we have to test everything too, to detect these :( See https://groups.google.com/d/msg/antlr-discussion/z8otgVzqB7I/IAHEU9keZLgJ